state/indexer/block.go

```diff
@@ -24,4 +24,10 @@ type BlockIndexer interface {
 	Search(ctx context.Context, q *query.Query) ([]int64, error)
 
 	SetLogger(l log.Logger)
+
+	Prune(retainHeight int64) (int64, int64, error)
```

Comment:
Since `BlockIndexer` is a public interface, I think we should add an extra changelog in the "breaking changes" directory to point at these changes

Reply:
Done: [.changelog/unreleased/breaking-changes/1176-blockindexer-pruning.md](https://github.com/cometbft/cometbft/blob/5b2522ff35ac86a28c703071c07bf72f9bd49124/.changelog/unreleased/breaking-changes/1176-blockindexer-pruning.md)